frontend/src/components/widgets/Multiselect/Multiselect.tsx

```diff
@@ -155,14 +172,25 @@ class Multiselect extends React.PureComponent<Props, State> {
   }
 
   private onChange = (params: OnChangeParams): void => {
-    const newState = this.generateNewState(params)
-    this.setState(newState, () => this.commitWidgetValue({ fromUi: true }))
+    if (
+      this.props.element.maxSelections &&
+      params.type === "select" &&
+      this.state.value.length >= this.props.element.maxSelections
+    ) {
+      return
+    }
+    this.setState(this.generateNewState(params), () =>
+      this.commitWidgetValue({ fromUi: true })
```

Comment:
nit: the callback function should be written

```javascript
() => { this.commitWidgetValue({ fromUi: true } }
```

to emphasize that `this.commitWidgetValue` is solely being run for its side effect and we don't care about its return value (there's no behavioral change in doing this, but it's semantically cleaner)